File: mobile/src/main/java/com/siliconlabs/bledemo/interfaces/DebugModeCallback.java
package com.siliconlabs.bledemo.interfaces;

import com.siliconlabs.bledemo.ble.BluetoothDeviceInfo;

public interface DebugModeCallback {

    void connectToDevice(BluetoothDeviceInfo device);

    void addToFavorite(String deviceAddress);

    void removeFromFavorite(String deviceAddress);

    void updateCountOfConnectedDevices();
}

File: mobile/src/main/java/com/siliconlabs/bledemo/features/scan/browser/adapters/DebugModeCallback.kt
package com.siliconlabs.bledemo.features.scan.browser.adapters

import android.bluetooth.BluetoothDevice
import com.siliconlabs.bledemo.bluetooth.ble.BluetoothDeviceInfo

interface DebugModeCallback {
    fun connectToDevice(position: Int, deviceInfo: BluetoothDeviceInfo)
    fun disconnectDevice(position: Int, device: BluetoothDevice)
    fun addToFavorites(deviceAddress: String)
    fun removeFromFavorites(deviceAddress: String)
    fun toggleViewExpansion(position: Int)
}
